cpp/lib/src/Downloader.cc

```diff
@@ -706,6 +679,38 @@ bool Downloader::getHttpEquivRedirect(std::string * const redirect_url) const {
 }
 
 
+void Downloader::setAcceptableLanguages(const std::string &acceptable_languages) {
+    params_.acceptable_languages_ = acceptable_languages;
+    RemoveAllWithMatchingPrefix(additional_http_headers_, "Accept-Language:");
+    additional_http_headers_ = curl_slist_append(additional_http_headers_,
+                                                 ("Accept-Language: " + params_.acceptable_languages_).c_str());
+}
+
+
+void Downloader::setIgnoreSslCertificates(const bool ignore_ssl_certificates) {
+    params_.ignore_ssl_certificates_ = ignore_ssl_certificates;
+    long curl_value(ignore_ssl_certificates ? 0L : 1L);
+    if (unlikely(::curl_easy_setopt(easy_handle_, CURLOPT_SSL_VERIFYPEER, curl_value) != CURLE_OK))
+        throw std::runtime_error("in Downloader::setIgnoreSslCertificates(): curl_easy_setopt() failed!");
+    if (unlikely(::curl_easy_setopt(easy_handle_, CURLOPT_SSL_VERIFYHOST, curl_value) != CURLE_OK))
+        throw std::runtime_error("in Downloader::setIgnoreSslCertificates(): curl_easy_setopt() failed!");
+}
+
+
+void Downloader::setProxy(const std::string &proxy_host_and_port) {
+    params_.proxy_host_and_port_ = proxy_host_and_port;
+    if (unlikely(::curl_easy_setopt(easy_handle_, CURLOPT_USERAGENT, proxy_host_and_port.c_str()) != CURLE_OK))
+        throw std::runtime_error("Downloader::setProxy(): curl_easy_setopt() failed! (" + proxy_host_and_port + ")");
+}
+
+
+void Downloader::setUserAgent(const std::string &user_agent) {
+    params_.user_agent_ = user_agent;
```

Comment:
Do we need to consider what we want to happen when "user_agent" is empty or don't we?

Reply:
Yes, see comment in Downloader.h